samples/snippets/src/main/java/com/example/storage/object/ComposeObject.java

```diff
@@ -50,10 +50,16 @@ public static void composeObject(
     // conditions and data corruptions. The request returns a 412 error if the
     // preconditions are not met.
     // For a target object that does not yet exist, set the DoesNotExist precondition.
+    // This will cause the request to fail if the object is created before the request runs.
     Storage.BlobTargetOption precondition = Storage.BlobTargetOption.doesNotExist();
     // If the destination already exists in your bucket, instead set a generation-match
-    // precondition:
-    // Storage.BlobTargetOption precondition = Storage.BlobTargetOption.generationMatch();
+    // precondition. This will cause the request to fail if the existing object's generation
+    // changes before the request runs.
+    if (storage.get(bucketName, targetObjectName) != null) {
```

Comment:
I think this makes more sense, because it will actually behave appropriately based on whether the object already exists or not. If it doesn't exist, the original precondition isn't overwritten

Reply:
Do you think it would look better as

```
Storage.BlobTargetOption precondition;
if (storage.get(bucketName, objectName != null) {
    precondition = ...;
} else {
    precondition = ...;
```
?

Reply:
Alternative that avoids the duplicate `storage.get` call:
```java
Storage.BlobTargetOption precondition = 
  Optional.ofNullable(storage.get(bucketName, targetObjectName))
    .map(Storage::getGeneration)
    .map(Storage.BlobTargetOption::generationMatch)
    .orElse(Storage.BlobTargetOption.doesNotExist())
```

Reply:
I think @JesseLovelace suggestion here is the most readable. If we wanted to avoid the duplicate call could we could save the object retrieved.